gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/Asset.java

```diff
@@ -50,12 +56,49 @@
 @API
 @ThreadSafe
 public abstract class Asset<T extends AssetData> {
+    private static final Logger logger = LoggerFactory.getLogger(Asset.class);
+
+    private final List<WeakReference<Asset<T>>> instances;
+    private Asset<T> parent;
 
     private final ResourceUrn urn;
     private final AssetType<?, T> assetType;
     private final DisposalHook disposalHook = new DisposalHook();
     private volatile boolean disposed;
 
+    private void appendInstance(Asset<T> asset) {
+        if (parent == null) {
+            Preconditions.checkArgument(!getUrn().isInstance());
+            asset.parent = this;
+            instances.add(new WeakReference<>(asset));
+        } else {
+            asset.parent = parent;
+            parent.instances.add(new WeakReference<>(asset));
+        }
+    }
+
+    protected Iterable<WeakReference<Asset<T>>> instances() {
```

Comment:
Can this be private/what reason is there for implementors to override this?

Reply:
Or call it.

Reply:
@pollend I think this question was not answered, was it? I see that you made it `final`, but why does it need to be on `protected` level?

Reply:
@pollend this question is still open

Reply:
you should be able to access all the list of instances from within the asset.